### PR TITLE
Implementing Form Validation

### PR DESCRIPTION
**Pull request by iamtmrobinson**
Basic client-side validation is now performed on both the register and comment edit/add forms.

* Converted the forms to classes to allow for keeping state and handling validation pre-submission.
* Renamed several functions in Panel and beyond to match the React convention of `onX={handleX}`, as well as to standardise on `Change` and `Submit` etc.

### CODE REVIEW

File: src/components/submitComment/index.js [hunk 1]
```diff
@@ -1,77 +1,116 @@
...
 
-  return (
-    <section className="blabbr-submitComment">
-      <form>
-        {formTitle}
-        <textarea
-          value={userComment}
-          onChange={onUserCommentChange}
-        />
+    this.handleChange = this.handleChange.bind(this);
```
Comment (shahzainb) [inline src/components/submitComment/index.js]: don't need this, see comment below

File: src/components/submitComment/index.js [hunk 2]
```diff
@@ -1,77 +1,116 @@
...
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
 
-        { type === 'Edit' ?
-        [
+  handleChange(event) {
```
Comment (shahzainb) [inline src/components/submitComment/index.js]: handleChange = (event) => {

then you don't need to bind it in constructor. cleaner.

File: src/components/submitComment/index.js [hunk 3]
```diff
@@ -1,77 +1,116 @@
...
+          <textarea
+            value={userComment}
+            onChange={this.handleChange}
+            ref="userComment"
+          />
+          { this.state.hasErrors && 
+            <div className="error">You need to enter a comment.</div>
+          }
+          { type === 'Edit' ?
+          [
```
Comment (shahzainb) [inline src/components/submitComment/index.js]: Maybe put this in some vars and a function or something? Looks a bit dirty right in the JSX.

**Review by shahzainb — CHANGES_REQUESTED** [review]

File: src/components/submitComment/index.js (hunk 2 above)
Comment (iamtmrobinson) [reply src/components/submitComment/index.js]: 👍  wasn't aware of this, nice. Have updated.

**Comment by iamtmrobinson** [comment]
@shahzainb have updated based on your comments, will save updating the main panel bind functions for another PR.

**Review by shahzainb — APPROVED** [review]

**Review by shahzainb — APPROVED** [review]

**Pull request merged**